homeassistant/components/weather/__init__.py

```diff
@@ -300,6 +300,31 @@ def __init_subclass__(cls, **kwargs: Any) -> None:
                     cls.__name__,
                     report_issue,
                 )
+        if any(method in cls.__dict__ for method in ("_attr_forecast", "forecast")):
```

Comment:
the code above will be removed once this PR has merged hence why code is duplicated